src/main.rs

```diff
@@ -8,57 +8,86 @@ use std::path::PathBuf;
+  Check {
+    #[arg(help = "Path to the input file")]
+    path: PathBuf,
+  },
+  /// Compiles the program to hvmc and prints to stdout.
+  Compile {
+    #[arg(
+      short = 'O',
+      value_delimiter = ' ',
+      action = clap::ArgAction::Append,
+      long_help = r#"Enables or disables the given optimizations
+      supercombinators is enabled by default."#,
+    )]
+    cli_opts: Vec<hvml::OptArgs>,
+
+    #[command(flatten)]
+    wopts: WOpts,
+
+    #[arg(help = "Path to the input file")]
+    path: PathBuf,
+  },
+  /// Compiles the program and runs it with the hvm.
+  Run {
+    #[arg(short, long, help = "How much memory to allocate for the runtime", default_value = "1G", value_parser = mem_parser)]
+    mem: usize,
 
-  #[arg(short, long, help = "Shows runtime stats and rewrite counts")]
-  pub stats: bool,
+    #[arg(short = 'd', help = "Debug mode (print each reduction step)")]
+    debug: bool,
 
-  #[arg(short, long, help = "How much memory to allocate for the runtime", default_value = "1G", value_parser = mem_parser)]
-  pub mem: usize,
+    #[arg(short = '1', help = "Single-core mode (no parallelism)")]
+    single_core: bool,
 
-  #[arg(short = '1', help = "Single-core mode (no parallelism)")]
-  pub single_core: bool,
+    #[arg(short = 'l', help = "Linear readback (show explicit dups)")]
+    linear: bool,
 
-  #[arg(short = 'd', help = "Debug mode (print each reduction step)")]
-  pub debug: bool,
+    #[arg(short, long = "stats", help = "Shows runtime stats and rewrite counts")]
+    arg_stats: bool,
 
-  #[arg(short = 'l', help = "Linear readback (show explicit dups)")]
-  pub linear: bool,
+    #[arg(help = "Path to the input file")]
+    path: PathBuf,
 
-  #[arg(
-    short = 'O',
-    value_delimiter = ' ',
-    action = clap::ArgAction::Append,
-    long_help = r#"Enables or disables the given optimizations
-    all, eta, no-eta, prune, no-prune, ref-to-ref, no-ref-to-ref,
-    supercombinators (enabled by default), no-supercombinators,
-    simplify-main, no-simplify-main, pre-reduce-refs, no-pre-reduce-refs
-    "#,
-  )]
-  pub opts: Vec<String>,
+    #[arg(
+      short = 'O',
+      value_delimiter = ' ',
+      action = clap::ArgAction::Append,
+      long_help = r#"Enables or disables the given optimizations
+      supercombinators is enabled by default."#,
+    )]
+    cli_opts: Vec<hvml::OptArgs>,
 
-  #[arg(help = "Path to the input file")]
-  pub path: PathBuf,
+    #[command(flatten)]
+    wopts: WOpts,
+  },
+  /// Runs the lambda-term level optimization passes.
+  Desugar {
+    #[arg(help = "Path to the input file")]
+    path: PathBuf,
+  },
 }
 
-#[derive(clap::Args, Debug)]
+#[derive(Args, Debug, Clone)]
 #[group(multiple = true)]
 struct WOpts {
   #[arg(
     short = 'W',
     long = "warn",
     value_delimiter = ' ',
     action = clap::ArgAction::Append,
-    long_help = r#"Show compilation warnings
-    all, unused-defs, match-only-vars"#
+    long_help = "Show compilation warnings",
```

Comment:
From just the help this is a bit ambiguous and seems more like it will show/allow/deny all warnings.

It would be better along the lines of "Show the specified compilation warning".

It would also be nice if the help command could show all the possible values somewhere and group these three options together somehow, eg:
"-W/D/A --warn/deny/allow : Show/Deny/Allow compilation warnings".
But I don't know how feasible that would be with clap